### PR TITLE
Fix address payload

### PR DESCRIPTION
**Pull request by apoorv1316**
Fixes 
https://saeloun-inc.sentry.io/issues/5250931858/?project=4504649746087936&query=is%3Aunresolved&referrer=issue-stream&stream_index=1

### CODE REVIEW

File: app/javascript/src/components/Clients/ClientForm/utils.ts [hunk 1]
```diff
@@ -50,7 +44,6 @@ export const formatFormData = (
 export const disableBtn = (values, errors, submitting) => {
   if (
     errors.name ||
-    errors.phone ||
```
Comment (nisusam) [inline app/javascript/src/components/Clients/ClientForm/utils.ts:53]: Is this relevant to this PR?

**Review by nisusam — APPROVED** [review]

**Review by nisusam — COMMENTED** [review]
This is still pointing to ruby 3.2.2.
Please rebase with develop once

File: app/javascript/src/components/Clients/ClientForm/utils.ts (hunk 1 above)
Comment (apoorv1316) [reply app/javascript/src/components/Clients/ClientForm/utils.ts:53]: Not related to this bug

File: app/javascript/src/components/Clients/ClientForm/utils.ts [hunk 2]
```diff
@@ -9,30 +9,24 @@ export const formatFormData = (
...
-    values.address1
-  );
-
-  formData.append(
-    "client[addresses_attributes[0][address_line_2]]",
-    values.address2
-  );
+  // eslint-disable-next-line no-nested-ternary
+  const addressIndex = isNewForm
+    ? 0
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Clients/ClientForm/utils.ts:14]: `const addressIndex = isNewForm ? 0 : client.address?.id ?? 0;`

File: app/javascript/src/components/Clients/ClientForm/utils.ts [hunk 3]
```diff
@@ -9,30 +9,24 @@ export const formatFormData = (
...
 
-  formData.append("client[addresses_attributes[0][state]]", values.state);
-
-  formData.append("client[addresses_attributes[0][city]]", values.city);
-
-  formData.append(
-    "client[addresses_attributes[0][country]]",
-    values.country?.value
-  );
+  if (!isNewForm && client.address && client.address.id) {
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Clients/ClientForm/utils.ts:20]:
```
if (!isNewForm && client?.address?.id) {
    formData.append(`${addressPrefix}[id]`, client.address.id);
}
```

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]
LGTM.
Small suggestions

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**